kaiax/reward/spec.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package reward
+
+import (
+	"math/big"
+
+	"github.com/kaiachain/kaia/common"
+)
+
+type RewardSummary struct {
+	Minted   *big.Int `json:"minted"`
+	TotalFee *big.Int `json:"totalFee"`
+	BurntFee *big.Int `json:"burntFee"`
+}
+
+func (summary *RewardSummary) Add(delta *RewardSummary) {
```

Comment:
Does it not need a nil check (`delta` and `properties` of `delta`)? 
Also consider `func (spec *RewardSpec) Add(delta *RewardSpec)` and `func (spec *RewardSpec) IncReceipient(addr common.Address, amount *big.Int)`.

Reply:
It's only for internal use and those use cases did not have nil pointers.
Let me make it private method `func (summary *RewardSummary) add(delta *RewardSummary)`.

Reply:
Fixed in [f472f15](https://github.com/kaiachain/kaia/pull/103/commits/f472f15ea194328bfa29faad9895bc035a196290). My approach is to making sure that every use case does not introduce a nil, not checking at runtime.